src/infra/src/table/folders.rs

```diff
@@ -126,8 +126,8 @@ pub async fn delete(org_id: &str, folder_id: &str) -> Result<(), errors::Error>
 }
 
 /// Gets a folder ORM entity by its `folder_id`.
-async fn get_model(
-    db: &DatabaseConnection,
+pub(crate) async fn get_model<C: ConnectionTrait>(
+    db: &C,
```

Comment:
Swap the concrete `DatabaseConection` type for the `ConnectionType` trait so we can pass in a transaction.